src/shared/components/plots/MultipleCategoryHeatmap.tsx

```diff
@@ -141,33 +140,38 @@ function toHeatmapTracks(
         []
     );
 
-    const totalSamplesPerGroup = result.map(t =>
-        t.data
-            .map(d => d.profile_data)
-            .reduce((partialSum: number, d: number) => partialSum + d, 0)
-    ) as number[];
+    const groupsPercentages: number[][] = [];
+
+    groups.forEach(group => {
+        const groupSampleCount =
+            group.data
+                .map(d => d.profile_data)
+                .reduce((partialSum: number, d: number) => partialSum + d, 0) ||
+            0;
```

Comment:
i don't understand this ```|| 0``` thing here

Reply:
in theory reduce can return null, so the `|| 0` is needed to fix typescript errors

Reply:
hmm. that doesn't sound right to me